cheroot/connections.py

```diff
@@ -99,76 +99,87 @@ def expire(self):
         # find any connections still registered with the selector
         # that have not been active recently enough.
         threshold = time.time() - self.server.timeout
-        timed_out_connections = (
+        timed_out_connections = [
             (sock_fd, conn)
             for _, (_, sock_fd, _, conn) in self._selector.get_map().items()
             if conn != self.server and conn.last_used < threshold
-        )
+        ]
         for sock_fd, conn in timed_out_connections:
             self._selector.unregister(sock_fd)
             conn.close()
 
-    def get_conn(self):  # noqa: C901  # FIXME
-        """Return a HTTPConnection object which is ready to be handled.
+    def stop(self):
+        """Stop the selector loop in run() synchronously.
 
-        A connection returned by this method should be ready for a worker
-        to handle it. If there are no connections ready, None will be
-        returned.
+        May take up to half a second.
```

Comment:
i am not aware of a cross-platform way to wake up the selector in response to an event - on Linux I'd typically use something like `eventfd` or create a pipe, but the selector on Windows can only monitor sockets.

Assuming that shutdown does not need to be optimized in practical usage, I figured it was easy enough to just check based on the selector poll timeout instead.